src/main/kotlin/snyk/container/ui/BaseImageRemediationDetailPanel.kt

```diff
@@ -0,0 +1,292 @@
+
+    init {
+        this.layout = GridLayoutManager(10, 1, Insets(20, 10, 20, 20), -1, 0)
+
+        this.add(
+            Spacer(),
+            baseGridConstraints(
+                9,
+                anchor = GridConstraints.ANCHOR_CENTER,
+                fill = GridConstraints.FILL_VERTICAL,
+                HSizePolicy = GridConstraints.SIZEPOLICY_CAN_SHRINK,
+                VSizePolicy = GridConstraints.SIZEPOLICY_WANT_GROW,
+                indent = 0
+            )
+        )
+
+        if (imageIssues.baseImageRemediationInfo != null) {
+            this.add(
+                baseRemediationInfoPanel(),
+                panelGridConstraints(8)
+            )
+        }
+
+        this.add(
+            getTitlePanel(),
+            panelGridConstraints(0)
+        )
+
+        this.name = "BaseImageRemediationDetailPanel"
+    }
+
+    private fun baseRemediationInfoPanel(): JPanel {
+        val panel = JPanel(GridLayoutManager(4, 4, Insets(20, 0, 0, 0), 0, 10))
+
+        panel.add(
+            JLabel("Recommendations for upgrading the base image"),
+            baseGridConstraints(0)
+        )
+
+        panel.add(
+            innerRemediationInfoPanel(),
+            baseGridConstraints(1)
+        )
+
+        return panel
+    }
+
+    private fun innerRemediationInfoPanel(): JPanel {
+        val panel = JPanel(GridLayoutManager(3, 4, Insets(0, 0, 0, 0), 0, 10))
+
+        imageIssues.baseImageRemediationInfo?.currentImage?.let {
+            addBaseImageInfo(panel, 0, CURRENT_IMAGE, it)
+        }
+
+        imageIssues.baseImageRemediationInfo?.minorUpgrades?.let {
+            addBaseImageInfo(panel, 1, MINOR_UPGRADES, it)
+        }
+
+        imageIssues.baseImageRemediationInfo?.alternativeUpgrades?.let {
+            addBaseImageInfo(panel, 2, ALTERNATIVE_UPGRADES, it)
+        }
+
+        return panel
+    }
+
+    private fun addBaseImageInfo(panel: JPanel, row: Int, title: String, info: BaseImageInfo) {
+        panel.add(
+            JLabel(title).apply {
+                font = io.snyk.plugin.ui.getFont(Font.BOLD, -1, JLabel().font)
+            },
+            baseGridConstraints(row, 0, indent = 0)
+        )
+        panel.add(
+            JLabel(info.name).apply {
+                name = title
+            },
+            baseGridConstraints(row, 1, indent = 5)
+        )
+        panel.add(
+            getVulnerabilitiesCHMLpanel(info.vulnerabilities),
+            baseGridConstraints(row, 2, indent = 5)
+        )
+    }
+
+    private fun panelGridConstraints(row: Int) = baseGridConstraints(
+        row = row,
+        anchor = GridConstraints.ANCHOR_CENTER,
+        fill = GridConstraints.FILL_BOTH,
+        HSizePolicy = GridConstraints.SIZEPOLICY_CAN_SHRINK or GridConstraints.SIZEPOLICY_CAN_GROW,
+        VSizePolicy = GridConstraints.SIZEPOLICY_CAN_SHRINK or GridConstraints.SIZEPOLICY_CAN_GROW,
+        indent = 0
+    )
+
+    private fun baseGridConstraints(
+        row: Int,
```

Comment:
Detekt - is this function really used with all parameters? Can we reduce the number of params?

Reply:
Would keep it as general as possible here. Also default parameter helps. 
But again we should refactor that and we actually should probably not use such constructor at all.